src/main/java/testProj/api/User.java

```diff
@@ -0,0 +1,36 @@
+package testProj.api;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Builder;
+
+import java.util.UUID;
+
+@Builder
+public class User {
```

Comment:
Please make fields final and use Lombok to generate getters. Constructors can be removed, an instance of User can be created using builder (and toBuilder() as replacement of copying constructor).
You can use @Jacksonized annotation to handle the serialization of immutable class with builder.